src/pages/index.js

```diff
@@ -5,12 +5,212 @@ import { PopupWithForm } from "../components/PopupWithForm.js";
 import { PopupWithImage } from "../components/PopupWithImage.js";
 import { Section } from "../components/Section.js";
 import { UserInfo } from "../components/UserInfo.js";
+import { Api } from "../components/api.js";
+import { PopupDeleteConfirm } from "../components/PopupDeleteConfirm.js";
 import utils from "../Utils/Constants.js";
 import "../pages/index.css";
 
-/*----------------------------------------------------------*/
-/*               			FUNCTIONS		                */
-/*----------------------------------------------------------*/
+//USER INFO
+const userInfo = new UserInfo({
+	titleSelector: utils.htmlIds.profileTitle,
+	descriptionSelector: utils.htmlIds.profileDescription,
+	profilePictureSelector: ".profile__image",
+});
+
+const deleteCardModal = new PopupDeleteConfirm("#js-card-delete-modal");
+
+//need to make a change picture pop up html
+const editProfilePictureForm = new PopupWithForm(
+	"#js-edit-profile-picture-modal",
+	handleEditProfilePicture
+);
+
+/*const cardDeleteButton = utils.cardTemplate.querySelector(
+	".card__delete-image"
+);*/
+
+/****** API CALLS***** */
+
+//CARD SECTION
+
+const opts = {
+	baseUrl: "https://around-api.en.tripleten-services.com/v1",
+	headers: {
+		authorization: "d50bb54b-1efc-4b8a-a5b1-3d5c72d6a1b0",
+		"Content-Type": "application/json",
+	},
+};
+let sectionCards;
+const api = new Api(opts);
+
+/** 1. GET ALL CARDS ROUTE */
+api.getInitialCards()
+	.then((data) => {
+		//handle the response data
+		console.log("Data from Api: ", data); //debugging statement
+
+		// Create a new Section instance with the fetched data and render the items
+		sectionCards = new Section(
+			{
+				items: data,
+				renderer: createCard,
+			},
+			utils.htmlIds.cardList
+		);
+		sectionCards.renderItems();
+	})
+	.catch((err) => {
+		console.log(err); //debugging statement
+		console.error(`Error getting cards: ${err} `);
+	});
+
+api.getUserInfo()
+	.then((res) => {
+		console.log("Inside getUserInfo, Response: ", res); //debugging
+		userInfo.setUserInfo({
+			name: res.name,
+			description: res.about,
+		});
+		userInfo.setProfilePicture({
+			profilePicture: res.avatar,
+		});
+	})
+	.catch((err) => {
+		console.log("ERROR: ", err);
+		console.error(err);
+	});
+
+/**
+ * Param: userData
+ * Description:	Sets the change/edit name/title and the description of the profile
+ * 					UPDATE USER PROFILE INFO ROUTE
+ */
+function handleProfileSubmitForm(userData) {
+	// Call the updateUserInfo method from the Api class
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```